src/Workspaces/CSharp/Portable/Workspace/LanguageServices/CSharpSyntaxTreeFactoryService.ParsedSyntaxTree.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Text;
+using System.Threading;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal partial class CSharpSyntaxTreeFactoryServiceFactory
+    {
+        private partial class CSharpSyntaxTreeFactoryService
+        {
+            /// <summary>
+            /// Parsed <see cref="CSharpSyntaxTree"/> that creates <see cref="SourceText"/> with given encoding and checksum algorithm.
+            /// </summary>
+            private sealed class ParsedSyntaxTree : CSharpSyntaxTree
```

Comment:
Functionally the ParsedSyntaxTree that the compiler defines is the same as this one. Since this is nested type I think it's actually better to have the same name.